packages/trezor/src/index.ts

```diff
@@ -0,0 +1,393 @@
+            throw new Error('There was an error accessing your Trezor accounts.')
+          }
+        }
+
+        function trezorSignTransactionLegacy(path: string, transactionData: any) {
+          const { nonce, gasPrice, gasLimit, gas, to, value, data } = transactionData
+      
+          return TrezorConnect.ethereumSignTransaction({
+            path: path,
+            transaction: {
+              to,
+              value: value || '',
+              data: data || '',
+              chainId: parseInt(currentChain?.id),
+              nonce,
+              gasPrice,
+              gasLimit: gasLimit || gas,
+            }
+          })
+        }
+
+        function trezorSignTransaction1559(path: string, transactionData: any) {
+          const { nonce, gas, gasLimit, to, value, data, maxFeePerGas, maxPriorityFeePerGas } = transactionData
+      
+          return TrezorConnect.ethereumSignTransaction({
+            path: path,
+            transaction: {
+              to,
+              value: value || '',
+              data: data || '',
+              chainId: parseInt(currentChain?.id),
+              nonce,
+              gasLimit: gasLimit || gas,
+              maxFeePerGas,
+              maxPriorityFeePerGas,
+            }
+          })
+        }
+
+        function trezorSignTransaction(path: string, transactionData: any) {
+          if (transactionData.hasOwnProperty('maxFeePerGas') || transactionData.hasOwnProperty('maxPriorityFeePerGas')) {
+            return trezorSignTransaction1559(path, transactionData)
+          }
+          return trezorSignTransactionLegacy(path, transactionData)
+        }
+      
+        async function signTransaction(transactionObject: any) {
+          if (!(accounts?.length && accounts?.length > 0))
+            throw new Error(
+              'No account selected. Must call eth_requestAccounts first.'
+            )
+          
+          const signingAccount =
+            accounts.find(
+              account => account.address === transactionObject?.from
+            ) || accounts[0]
+
+          const { address: from, derivationPath } = signingAccount
+
+          // Set the `from` field to the currently selected account
+          transactionObject = { ...transactionObject, from }
+
+          const common = new Common({
+            chain: customNetwork || Number.parseInt(currentChain?.id) || 1,
+            // Berlin is the minimum hardfork that will allow for EIP1559
+            hardfork: Hardfork.Berlin,
+            // List of supported EIPS
+            eips: [1559]
+          })
+          
+          const { BN, toBuffer } = ethUtil
+          
+          const transaction = Transaction.fromTxData(
+            {
+              ...transactionObject,
+              gasLimit: transactionObject.gas ?? transactionObject.gasLimit
+            },
+            { common, freeze: false }
+            )
+
+          transaction.v = new BN(toBuffer(currentChain?.id))
```

Comment:
I am seeing TypeScript errors here from writing over a read only parameter that we will need to resolve.